src/controllers/opportunities.js

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright 2023 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {
+  badRequest,
+  notFound,
+  ok,
+  createResponse,
+  noContent,
+} from '@adobe/spacecat-shared-http-utils';
+import {
+  hasText,
+  isObject,
+  isNonEmptyObject,
+  arrayEquals,
+} from '@adobe/spacecat-shared-utils';
+
+import { OpportunityDto } from '../dto/opportunity.js';
+
+/**
+ * Opportunities controller.
+ * @param {DataAccess} dataAccess - Data access.
+ * @returns {object} Opportunities controller.
+ * @constructor
+ */
+function OpportunitiesController(dataAccess) {
+  if (!isObject(dataAccess)) {
+    throw new Error('Data access required');
+  }
+  const { Opportunity } = dataAccess;
+  if (!isObject(Opportunity)) {
+    throw new Error('Data access required');
+  }
+
+  /**
+   * Gets all opportunities for a given site.
+   * @param {Object} context of the request
+   * @returns {Promise<Response>} Array of opportunities response.
+   */
+  const getAllForSite = async (context) => {
+    const siteId = context.params?.siteId;
+
+    if (!hasText(siteId)) {
+      return badRequest('Site ID required');
+    }
+
+    const opptys = (await Opportunity.allBySiteId(siteId))
+      .map((oppty) => OpportunityDto.toJSON(oppty));
+
+    return ok(opptys);
+  };
+
+  /**
+   * Gets all opportunities for a given site type filtering by status.
+   * @param {Object} context of the request
+   * @returns {Promise<Response>} Array of opportunities response.
+   */
+  const getByStatus = async (context) => {
+    const siteId = context.params?.siteId;
+    const status = context.params?.status || undefined;
+
+    if (!hasText(siteId)) {
+      return badRequest('Site ID required');
+    }
+    if (!hasText(status)) {
+      return badRequest('Status required');
+    }
+
+    const opptys = (await Opportunity.allBySiteIdAndStatus(siteId, status))
+      .map((oppty) => OpportunityDto.toJSON(oppty));
+
+    return ok(opptys);
+  };
+
+  /**
+   * Gets an opportunity for a given site type and opportunity ID.
+   * @param {Object} context of the request
+   * @returns {Promise<Response>} Opportunity response.
+   */
+  const getByID = async (context) => {
+    const siteId = context.params?.siteId;
+    const opptyId = context.params?.opportunityId || undefined;
```

Comment:
why is this branch needed?

Reply:
it is not needed. Side effects of copy-pasting. 
Removing it